gusto/kernels.py

```diff
@@ -0,0 +1,420 @@
+"""
+This file provides kernels for par loops.
+This are contained in this file as functions so that they can be tested separately.
+"""
+
+import numpy as np
+from firedrake import dx
+from firedrake.parloops import par_loop, READ, INC, WRITE
+from pyop2 import ON_TOP, ON_BOTTOM
+
+
+class GaussianElimination(object):
+    """
+    A kernel for performing Gaussian elimination locally in each element
+    for the BoundaryRecoverer procedure.
+
+    This takes a scalar field in DG1, the coordinates of the DoFs and the
+    "effective" coordinates of the DoFs. These "effective" coordinates
```

Comment:
If I understand correctly, this describes more input args than the `__init__` method takes.

Reply:
I've updated this comment to make it clearer